core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -372,6 +373,7 @@ public PlanOptimizers(
                 .addAll(new RemoveRedundantDateTrunc(plannerContext, typeAnalyzer).rules())
                 .addAll(new ArraySortAfterArrayDistinct(plannerContext).rules())
                 .add(new RemoveTrivialFilters())
+                .add(new RemoveFalseFiltersAfterDomainTranslator(plannerContext))
```

Comment:
RemoveTrivialFilters is used twice, the new rule is used once. what's the selection criteria to when to use the new rule and when not to?

Reply:
Good question, I wanted to avoid `PushPredicateIntoTableScan` from getting into this `if` clause 
```
if (newDomain.isNone()) {
            // This is just for extra safety, the rule RemoveFalseFiltersAfterDomainTranslator is responsible for eliminating such filters
```
I saw that we run `simplifyOptimizer` at line 555 and then `PushPredicateIntoTableScan` for the first time at line 598 so I thought it was not needed at line 438 as well. But thinking about it again, if we have a chance of getting a trivial filter at line 438, maybe it's worth removing `FALSE` filters as well at this point. Added.

Reply:
I wanted to avoid `PushPredicateIntoTableScan` from getting into this `if` clause 
```
if (newDomain.isNone()) {
            // This is just for extra safety, the rule RemoveFalseFiltersAfterDomainTranslator is responsible for eliminating such filters
```
I saw that we run `simplifyOptimizer` at line 556 and then `PushPredicateIntoTableScan` for the first time at line 600 so I thought it was not needed at line 438 as well. But thinking about it again, if we have a chance of getting a trivial filter at line 438, maybe it's worth removing `FALSE` filters at this point as well. Added.